src/utils/field-utils.js

```diff
@@ -0,0 +1,21 @@
+/* eslint import/prefer-default-export: "off" */
+
+/**
+ * Retrieves the model to which the supplied field refers.
+ *
+ * @param {Object} field - The field definition to determine the ref for
+ * @returns {Model} - The model to which the field refers
+ */
+export function getRef(field) {
+    if (!field.ref) {
+        return null;
+    }
+
+    if (field.ref._modelMeta) {
+        return field.ref;
+    }
+    if (Object.prototype.toString.call(field.ref) === '[object Function]') {
```

Comment:
To clarify for anyone who may read this who wasn't involved in the face-to-face conversation, calling the `toString` prototype function avoids calling a method on undefined. Assuming we somehow make it past the first check for if the ref doesn't exist, doing a `null.toString()` or `undefined.toString()` type of thing would yield a TypeError, which we avoid here. Force of habit. IIRC it's one of the lodash or underscore guys that recommends this in the absence of either. It's unlikely to be the most performant way to handle the situation, but I'm willing to forsake a wee performance gain in favour of readable code, save the need for greater performance arises.